### ISSUE TITLE
Raspberry Pi SSH over Bluetooth

### ISSUE THREAD

**Issue by blackfunnyduck** [bablokb/pi-btnap#7]
I am struggling with finding a way to ssh into pi via bluetooth and I found this repository
Although I carefully followed all the steps to get the btnap server working, there are two missing packages that cannot be installed, maybe due to deprecation.
When running ./pi-btnap/tools/install-btnap server it returns these errors:

Package python-dbus is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package python-gobject is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-dbus' has no installation candidate
E: Package 'python-gobject' has no installation candidate

I didn't find a way to get these missing packages installed as to be used by the install script.
Your kind suggestions on how to get around this would be much appreciated.

My system is Raspberry pi zero 2 w running Raspberry Pi OS 32 bit kernel 5.10.92-v7+

**Comment by bablokb** [comment]
yes, these are python2 packages and python2 is deprecated. So it would be time to port the code to python3 (python3 has similar packages). The bigger problem might be the bluetooth dbus interface, this might or might not have also changed.

So it is hard to judge how much work it would be to update the scripts to a modern python3/bluez-system.

I know that I won't find the time for this, but will of course accept patches.

As a workaround if you only need a shell and not full network-functions: why don't you use rfcomm serial interface?

**Comment by blackfunnyduck** [comment]
> yes, these are python2 packages and python2 is deprecated. So it would be time to port the code to python3 (python3 has similar packages). The bigger problem might be the bluetooth dbus interface, this might or might not have also changed.
> 
> So it is hard to judge how much work it would be to update the scripts to a modern python3/bluez-system.
> 
> I know that I won't find the time for this, but will of course accept patches.
> 
> As a workaround if you only need a shell and not full network-functions: why don't you use rfcomm serial interface?

Thank you so much for the swift reply.
Indeed I need just a shell to control the pi via bluetooth.
Although I am not familiar with bluetooth rfcomm it looks like the reasonable thing to do so I will give it a try.
